From: Federico Pellegrin <fede@evolware.org>
Date: Thu, 27 Jan 2022 03:36:36 +0100
Subject: [PATCH] gpmi-nand fix MX6 clock bug from upstream

Apply patch to fix problems with some Micron chips, see:

https://github.com/torvalds/linux/commit/aa1baa0e6c1aa4872e481dce4fc7fd6f3dd8496b#diff-0dec2fa8640ea2067789c406ab1e42c9805d0d0fc9f70a3a29d17f9311e23ca2

and discussions at:

https://lore.kernel.org/all/4352240.LvFx2qVVIh@n95hx1g2/

diff -ruN a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
--- a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c	2021-10-30 11:27:44.872933754 +0200
+++ b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c	2022-01-27 03:36:39.734745302 +0100
@@ -1184,15 +1184,6 @@
 		r->clock[i] = clk;
 	}
 
-	if (GPMI_IS_MX6(this))
-		/*
-		 * Set the default value for the gpmi clock.
-		 *
-		 * If you want to use the ONFI nand which is in the
-		 * Synchronous Mode, you should change the clock as you need.
-		 */
-		clk_set_rate(r->clock[0], 22000000);
-
 	return 0;
 
 err_clock:
